src/nvim/lib/kbtree.h

```diff
@@ -0,0 +1,430 @@
+/*-
+ * Copyright 1997-1999, 2001, John-Mark Gurney.
+ *           2008-2009, Attractive Chaos <attractor@live.co.uk>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#ifndef NVIM_LIB_KBTREE_H
+#define NVIM_LIB_KBTREE_H
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+#define KB_MAX_DEPTH 64
+
+#define	__KB_KEY(type, x)	(x->key)
+#define __KB_PTR(btr, x)	(x->ptr)
+
+#define __KB_TREE_T(name,key_t,T)						\
+    typedef struct kbnode_##name##_s kbnode_##name##_t;     \
+    struct kbnode_##name##_s {              \
+      int32_t n; \
+      bool is_internal; \
+      key_t key[2*T-1]; \
+      kbnode_##name##_t *ptr[]; \
+    } ; \
+    \
+    typedef struct { \
+        kbnode_##name##_t *root; \
+        int	n_keys, n_nodes; \
+    } kbtree_##name##_t; \
+    \
+    typedef struct { \
+        kbnode_##name##_t *x; \
+        int i; \
+    } kbpos_##name##_t; \
+    typedef struct { \
+        kbpos_##name##_t stack[KB_MAX_DEPTH], *p; \
+    } kbitr_##name##_t; \
+
+
+#define __kb_destroy(kbnode_t,b) do {											\
+		int i;                                                          \
+        unsigned int max = 8;											\
+		kbnode_t *x, **top, **stack = 0;								\
+		if (b->root) {													\
+			top = stack = (kbnode_t**)calloc(max, sizeof(kbnode_t*));	\
+			*top++ = (b)->root;											\
+			while (top != stack) {										\
+				x = *--top;												\
+				if (x->is_internal == 0) { free(x); continue; }			\
+				for (i = 0; i <= x->n; ++i)								\
+					if (__KB_PTR(b, x)[i]) {							\
+						if (top - stack == (int)max) {		        	\
+							max <<= 1;									\
+							stack = (kbnode_t**)realloc(stack, max * sizeof(kbnode_t*)); \
```

Comment:
:+1: Fixed.

Because the file like the other `lib/`-s is full of linter errors.